pyquil/device/_main.py

```diff
@@ -123,26 +129,71 @@ def get_isa(self, oneq_type=None, twoq_type=None) -> ISA:
             raise ValueError("oneq_type and twoq_type are both fatally deprecated. If you want to "
                              "make an ISA with custom gate types, you'll have to do it by hand.")
 
-        qubits = [Qubit(id=q.id, type=None, dead=q.dead, gates=[
-            MeasureInfo(operator="MEASURE", qubit=q.id, target="_",
-                        fidelity=self.specs.fROs()[q.id] or DEFAULT_MEASURE_FIDELITY,
-                        duration=DEFAULT_MEASURE_DURATION),
-            MeasureInfo(operator="MEASURE", qubit=q.id, target=None,
-                        fidelity=self.specs.fROs()[q.id] or DEFAULT_MEASURE_FIDELITY,
-                        duration=DEFAULT_MEASURE_DURATION),
-            GateInfo(operator="RZ", parameters=["_"], arguments=[q.id],
-                     duration=PERFECT_DURATION, fidelity=PERFECT_FIDELITY),
-            GateInfo(operator="RX", parameters=[0.0], arguments=[q.id],
-                     duration=DEFAULT_RX_DURATION, fidelity=PERFECT_FIDELITY)] + [
-                GateInfo(operator="RX", parameters=[param], arguments=[q.id],
-                         duration=DEFAULT_RX_DURATION,
-                         fidelity=self.specs.f1QRBs()[q.id] or DEFAULT_RX_FIDELITY)
-                for param in [np.pi, -np.pi, np.pi / 2, -np.pi / 2]])
+        def safely_get(attr, index, default):
+            if self.specs is None:
+                return default
+
+            getter = getattr(self.specs, attr, None)
+            if getter is None:
+                return default
+
+            array = getter()
+            if index < len(array):
+                return array[index]
+            else:
+                return default
```

Comment:
this causes an error for me. I think just making it 
`return array.get(index, default)`
works